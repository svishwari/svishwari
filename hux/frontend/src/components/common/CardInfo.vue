<template>
  <v-hover>
    <template v-slot:default="{ hover }">
      <v-card
        class="text-center rounded-lg"
        min-width="220px"
        max-width="260px"
        :disabled="!active"
        :to="{ name: to }"
        @click="$emit('click')"
        :elevation="hover ? 10 : 0"
        :hover="true"
        :outlined="!active"
      >
        <v-btn fab :ripple="false" elevation="0" outlined color="zircon" class="mt-4">
          <v-icon color="primary" x-large>
            {{ icon }}
          </v-icon>
        </v-btn>
        <v-card-title class="justify-center font-weight-regular" v-bind:class="{ 'primary--text': active }">
          {{ title }}
        </v-card-title>
        <v-card-subtitle v-bind:class="{ 'neroBlack--text': active }">
          {{ description }}
        </v-card-subtitle>
      </v-card>
    </template>
  </v-hover>
</template>

<script>
export default {
  name: "card-info",

  props: {
    icon: {
      type: String,
      required: false,
      default: "mdi-plus",
    },

    title: {
      type: String,
      required: false,
      default: "Info card title",
    },

    description: {
      type: String,
      required: false,
      default: "Info card description",
    },

    active: {
      type: Boolean,
      required: false,
      default: true,
    },

    to: {
      type: String,
      required: false,
    },
  },
}
</script>
