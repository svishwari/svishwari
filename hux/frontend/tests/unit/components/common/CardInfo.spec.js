import CardInfo from "@/components/common/CardInfo"
import { shallowMount } from "@vue/test-utils"

describe("Card Info", () => {
  test("Card info displays a default", () => {
    const defaultProps = {
      icon: "mdi-plus",
      title: "Info card title",
      description: "Info card description",
    }
    const wrapper = shallowMount(CardInfo)

    expect(wrapper.text()).toContain(defaultProps.title)
    expect(wrapper.text()).toContain(defaultProps.description)
    expect(wrapper).toMatchSnapshot()
  })

  test("Card info displays custom properties", () => {
    const customProps = {
      icon: "mdi-facebook",
      title: "Add a Destination",
      description:
        "Choose a destination where your actionable intelligence will be consumed.",
    }

    const wrapper = shallowMount(CardInfo, {
      propsData: customProps,
    })

    expect(wrapper.text()).toContain(customProps.title)
    expect(wrapper.text()).toContain(customProps.description)
    expect(wrapper).toMatchSnapshot()
  })
})
