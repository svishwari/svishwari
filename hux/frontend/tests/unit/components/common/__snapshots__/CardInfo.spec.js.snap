// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Info Card info displays a default 1`] = `
<v-hover-stub opendelay="0" closedelay="0">
  <v-card-stub loaderheight="4" tag="div" elevation="0" maxwidth="260px" minwidth="220px" hover="true" class="text-center rounded-lg card-info-wrapper card-shadow pa-3">
    <v-btn-stub color="zircon" elevation="0" outlined="true" tag="button" activeclass="" fab="true" type="button" class="mt-4">
      <v-icon-stub color="primary" xlarge="" x-large="">
        mdi-plus
      </v-icon-stub>
    </v-btn-stub>
    <v-card-title-stub datae2e="card-title" class="justify-center font-weight-regular primary--text">
      Info card title
    </v-card-title-stub>
    <v-card-subtitle-stub datae2e="card-description" class="black--text text--darken-4">
      Info card description
    </v-card-subtitle-stub>
  </v-card-stub>
</v-hover-stub>
`;

exports[`Card Info Card info displays custom properties 1`] = `
<v-hover-stub opendelay="0" closedelay="0">
  <v-card-stub loaderheight="4" tag="div" elevation="0" maxwidth="260px" minwidth="220px" hover="true" class="text-center rounded-lg card-info-wrapper card-shadow pa-3">
    <v-btn-stub color="zircon" elevation="0" outlined="true" tag="button" activeclass="" fab="true" type="button" class="mt-4">
      <v-icon-stub color="primary" xlarge="" x-large="">
        mdi-facebook
      </v-icon-stub>
    </v-btn-stub>
    <v-card-title-stub datae2e="card-title" class="justify-center font-weight-regular primary--text">
      Add a Destination
    </v-card-title-stub>
    <v-card-subtitle-stub datae2e="card-description" class="black--text text--darken-4">
      Choose a destination where your actionable intelligence will be consumed.
    </v-card-subtitle-stub>
  </v-card-stub>
</v-hover-stub>
`;
